packages/dev/src/main.ts

```diff
@@ -486,11 +514,17 @@ export class NetlifyDev {
         ),
       }
 
+      const geolocation = await getGeoLocation({
```

Comment:
let's share the instance across functions and edge functions. I think we can basically just `let geolocation` outside the conditional scopes above and then use `??=` in both these spots? something like that. keep it simple and make sure TS is happy without hacks.